### PR TITLE
Update anti-request-forgery.md

### LINKED ISSUE #25564: Clarify antiforgery protection behavior around multiple tabs being open
Initially I used the "Synchronizer Token Pattern (STP)" pattern described in the docs under "Antiforgery in ASP.NET Core". This led to issues for any users that use my website from multiple tabs, due to STP enforcing page load order (only POST made by the most recently loaded page would work properly).

I think the documentation could be improved by explaining more about this behavior, and recommending that in cases where you don't want to enforce page load ordering, the cookie to header pattern, (also documented but not named) or another CSRF protection pattern, may be suitable. I am happy to make this change myself.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 47b14f57-82ac-a2e2-cbc7-22a81a60f4ac
* Version Independent ID: bffca13c-223f-c61f-9cb2-9da8811eecfa
* Content: [Prevent Cross-Site Request Forgery (XSRF/CSRF) attacks in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-6.0)
* Content Source: [aspnetcore/security/anti-request-forgery.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/anti-request-forgery.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 350915](https://dev.azure.com/msft-skilling/Content/_workitems/edit/350915)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #25564

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/anti-request-forgery.md](https://github.com/dotnet/AspNetCore.Docs/blob/7f1460b765bb87db688b673bbd5236194a66eda0/aspnetcore/security/anti-request-forgery.md) | [aspnetcore/security/anti-request-forgery](https://review.learn.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?branch=pr-en-us-34257) |

### CODE REVIEW

File: aspnetcore/security/anti-request-forgery.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ Calling <xref:Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExten
 
 ## Multiple browser tabs and the Synchronizer Token Pattern
 
-With the Synchronizer Token Pattern, only the most recently loaded page contains a valid antiforgery token. Using multiple tabs can be problematic. For example, if a user opens multiple tabs:
+With the Synchronizer Token Pattern, only the most recently loaded page is guaranteed to contain a valid antiforgery token. ***Using multiple tabs can be problematic***. For example, if a user opens multiple tabs:
```
Comment (Rick-Anderson) [inline aspnetcore/security/anti-request-forgery.md:162]:
```suggestion
With the Synchronizer Token Pattern, only the most recently loaded page is guaranteed to contain a valid antiforgery token. Apps that wish to support multiple tabs should test supported browser and log failures. ***Using multiple tabs can be problematic***. For example, if a user opens multiple tabs:
```

File: aspnetcore/security/anti-request-forgery.md [hunk 2]
```diff
@@ -159,7 +159,7 @@ Calling <xref:Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExten
 
 ## Multiple browser tabs and the Synchronizer Token Pattern
 
-With the Synchronizer Token Pattern, only the most recently loaded page contains a valid antiforgery token. Using multiple tabs can be problematic. For example, if a user opens multiple tabs:
+With the Synchronizer Token Pattern, only the most recently loaded page is guaranteed to contain a valid antiforgery token. Apps that wish to support multiple tabs should test supported browsers and log failures. ***Using multiple tabs can be problematic***. For example, if a user opens multiple tabs:
```
Comment (tdykstra) [inline aspnetcore/security/anti-request-forgery.md:162]: The intro to the bulleted list is now out of sync with the list itself:

* is guaranteed to contain
* contains

Consider changing the second bullet with something similar, like "might fail with an error"

Suggestion that you can accept if you want is in the next comment

File: aspnetcore/security/anti-request-forgery.md [hunk 3]
```diff
@@ -159,7 +159,7 @@ Calling <xref:Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExten
...
 
  * Only the most recently loaded tab contains a valid antiforgery token.
  * Requests made from previously loaded tabs fail with an error: `Antiforgery token validation failed. The antiforgery cookie token and request token do not match`
```
Comment (tdykstra) [inline aspnetcore/security/anti-request-forgery.md:165]:
```suggestion
 * Only the most recently loaded tab is guaranteed to contain a valid antiforgery token.
 * Requests made from previously loaded tabs might fail with an error: `Antiforgery token validation failed. The antiforgery cookie token and request token do not match`
```

**Review by tdykstra — APPROVED** [review]
Left a suggestion.

**Pull request merged**